### ISSUE TITLE
False positive: Null pointer exception => null coalesce operator on object properties.

### ISSUE THREAD

**Issue by thomasphansen** [kalessil/phpinspectionsea#952]
PHP Inspections (EA Extended) marks the following code as "Null pointer exception may occur here":
```
$a = $test->mytest ?? 0;
```
, even though this will never throw an exception (since the null coalesce operator will handle it).

This is valid PHP syntax (and, IMHO, one of the reasons why the null coalesce operator was created for). An analysis on it can be found in [this article](http://markshust.com/2017/08/25/php-7s-null-coalesce-operator-usage-objects-and-arrays), as quoted below:

> What isn't documented anywhere though is how you can benefit from null coalesce when working with objects and arrays. Let's say you are wanting to return a property on an object, but aren't sure if that object exists. You need to check the base object by doing multiple checks on object properties and their children:
>```
>class Foo {}
>$foo = new Foo;
>echo isset($foo) && isset($foo->b) && isset($foo->b->c) ? $foo->b->c : 'baz';
>```
>You can see how it can easily become a handful when working with many object properties. With null coalesce, all you need to do is call the deepest object property directly. This code is executes exactly the same as the above:
>```
>class Foo {}
>$foo = new Foo;
>echo $foo->b->c ?? 'baz';
>```
>And it even works if the object isn't even defined. This script runs without errors:
>```
>echo $foo->b->c ?? 'baz';
>```

Please notice: this is only true for **properties**: a call to a method on an non-existent/null variable or class _will_ throw an exception, no matter if the null coalesce operator is present:
```
$a = $test->mytest() ?? 0; // this will fail
```

**Comment by kalessil** [comment]
Looks like a bug, thank you @thomasphansen

**Comment by kalessil** [comment]
Fixed
